src/analysis/split.jl

```diff
@@ -49,42 +62,57 @@ function split_function_head(ex::Expr; source=nothing)
         Expr(:block, x, ::LineNumberNode, Expr(:(=), kw, value)) => (nothing, Any[x], Any[Expr(:kw, kw, value)], nothing, nothing)
         Expr(:block, x, ::LineNumberNode, kw) => (nothing, Any[x], Any[kw], nothing, nothing)
         Expr(:(::), call::Expr, rettype) => begin
-            name, args, kw, whereparams, _ = split_function_head(call)
+            sub_tuple = _split_function_head_nothrow(call)
+            isnothing(sub_tuple) && return nothing
+            name, args, kw, whereparams, _ = _split_function_head_nothrow(call)
             (name, args, kw, whereparams, rettype)
         end
         Expr(:where, call, whereparams...) => begin
-            name, args, kw, _, rettype = split_function_head(call)
+            sub_tuple = _split_function_head_nothrow(call)
+            isnothing(sub_tuple) && return nothing
+            name, args, kw, _, rettype = sub_tuple
             (name, args, kw, whereparams, rettype)
         end
-        _ => throw(SyntaxError("expect a function head, got $ex", source))
+        _ => nothing
     end
 end
-split_function_head(s::Symbol; source=nothing) = (nothing, Any[s], nothing, nothing, nothing)
+_split_function_head_nothrow(s::Symbol) = (nothing, Any[s], nothing, nothing, nothing)
 
 """
     split_anonymous_function_head(ex::Expr) -> nothing, args, kw, whereparams, rettype
 
 Split anonymous function head to arguments, keyword arguments and where parameters.
 """
 function split_anonymous_function_head(ex::Expr; source=nothing)
+    split_head_tuple = _split_function_head_nothrow(ex)
+    isnothing(split_head_tuple) && throw(SyntaxError("expect an anonymous function head, got $ex", source))
+    split_head_tuple
+end
+
+function _split_anonymous_function_head_nothrow(ex::Expr)
```

Comment:
```suggestion
function split_anonymous_function_head_nothrow(ex::Expr)
```

just make it an API since maybe there is a use case for nothrow functions?